crates/ruff_python_trivia/src/textwrap.rs

```diff
@@ -122,6 +124,51 @@ pub fn dedent(text: &str) -> Cow<'_, str> {
     Cow::Owned(result)
 }
 
+/// Reduce a block's indentation to match the provided indentation.
+///
+/// This function looks at the first line in the block to determine the
+/// current indentation, then removes whitespace from each line to
+/// match the provided indentation.
+///
+/// Lines that are indented by _less_ than the indent of the first line
+/// are left unchanged.
+///
+/// Lines that consist solely of whitespace are trimmed to a blank line.
+///
+/// # Panics
+/// If the first line is indented by less than the provided indent.
+pub fn dedent_to(text: &str, indent: &str) -> String {
+    // Look at the indentation of the first line, to determine the "baseline" indentation.
+    let existing_indent_len = text
+        .universal_newlines()
+        .next()
+        .map_or(0, |line| line.len() - line.trim_start().len());
```

Comment:
Nit: Why not `indentation_at_offset`?

Reply:
Because the text here is the "full line", not the start of the content.